mautic/mautic
# Show publish icons only when having proper permissions.

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | #3781 
 BC breaks? | 
| Deprecations? | 

[//]: # ( Required: )
#### Description:
The publish button was displayed even in cases user did not have proper permissions. This PR fixes this.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a new user without any permissions.
2. For every item in left menu (Assets, Forms, Marketing Messages, etc...) set the new user's permissions to View own/View Other/View.
3. For every item in left menu, the publish button can be still seen, although it does nothing when clicked (a spinning wheel animation pops up indefinitely).

#### Steps to test this PR:
1. Apply this PR
2. Repeat reproduce steps
3. The publish button should not be visible now.

#### Backwards compatibility breaks:
I've modified the `app/bundles/CoreBundle/Views/Standard/list.html.php` file. This is apparently used only for Marketing Messages, but if anyone else uses this "default" template, they might have some problems, since the template now checks for `permissions[$permissionBase.':publish']`.
